classes/class-ppc-pagesetups.php

```diff
@@ -0,0 +1,257 @@
+			if ( ! isset( $ppc_screen->parent_base ) || ( isset( $ppc_screen->parent_base ) && 'edit' !== $ppc_screen->parent_base ) ) {
+				return;
+			}
+			wp_enqueue_script( 'ppc_backend_checkbox_js' );
+			wp_enqueue_style( 'ppc_backend_css' );
+			?>
+			<div class = "ppc-modal-warn">
+				<div id="ppc_notifications" class="ppc-popup-warn">
+					<h2>Pre-Publish Checklist</h2>
+					<p class="ppc-popup-description">Your Pre-Publish Checklist is incomplete. What would you like to do?</p>
+					<div class="ppc-button-wrapper">
+						<div class="ppc-popup-option-dontpublish">Don't Publish</div>
+						<div class="ppc-popup-options-publishanyway">Publish Anyway</div>
+					</div>    
+				</div>
+			</div>
+			<div class = "ppc-modal-prevent">
+				<div id="ppc_notifications" class="ppc-popup-prevent">
+					<h2>Pre-Publish Checklist</h2>
+					<p class="ppc-popup-description"> Please check all the checklist items before publishing.</p>
+					<div class="ppc-prevent-button-wrapper">
+						<div class="ppc-popup-option-okay">Okay, Take Me to the List!</div>
+					</div>  	
+				</div>
+			</div>
+			<?php
+		}
+
+		/**
+		 * Function for adding settings page in admin area
+		 *
+		 * Displays our plugin settings page in the WordPress
+		 *
+		 * @since 1.0.0
+		 */
+		public function ppc_settings_page() {
+			add_submenu_page(
+				'options-general.php',
+				'Pre-Publish Checklist',
+				'Pre-Publish Checklist',
+				'manage_options',
+				'ppc',
+				array( $this, 'ppc_page_html' )
+			);
+		}
+		/**
+		 * Tabs function
+		 *
+		 * All the tabs are managed in the file which is included.
+		 *
+		 * @since 1.0.0
+		 */
+		public function ppc_page_html() {
+			require_once PPC_ABSPATH . 'includes/ppc-tabs.php';
+		}
+		/**
+		 * Add custom meta box
+		 *
+		 * Display plugin's custom meta box in the meta settings side bar
+		 *
+		 * @since 1.0.0
+		 */
+		public function ppc_add_custom_meta_box() {
+			$ppc_post_types_to_display = get_option( 'ppc_post_types_to_display' );
+			if ( ! empty( $ppc_post_types_to_display ) || false !== $ppc_post_types_to_display ) {
+				foreach ( $ppc_post_types_to_display as $screen ) {
+					add_meta_box(
+						'ppc_custom_meta_box', // Unique ID.
+						'Pre-Publish Checklist', // Box title.
+						array( $this, 'ppc_custom_box_html' ), // Content callback, must be of type callable.
+						$screen,
+						'side',
+						'high'
+					);
+				}
+			}
+		}
+
+		/**
+		 * Call back function for HTML markup of meta box.
+		 *
+		 * This functions contains the markup to be displayed or the information to be displayed in the custom meta box
+		 *
+		 * @since 1.0.0
+		 */
+		public function ppc_custom_box_html() {
+			wp_enqueue_script( 'ppc_backend_checkbox_js' );
+			wp_enqueue_script( 'ppc_backend_tooltip_js' );
+			wp_enqueue_style( 'ppc_backend_css' );
+			global $post;
+			$ppc_checklist_item_data = get_option( 'ppc_checklist_data' );
+			$value                   = get_post_meta( $post->ID, '_ppc_meta_key', true );
```

Comment:
Instead of using `global $post` prefer using [get_the_id()](https://developer.wordpress.org/reference/functions/get_the_id/)